utils/emoji.go

```diff
@@ -60,3 +67,29 @@ func CreateTestPng(t *testing.T, width int, height int) []byte {
 
 	return buffer.Bytes()
 }
+
+func GetEmoji(emoji string) string {
+	emojiVal, ok := model.SystemEmojis[strings.Trim(emoji, ":")]
+	if !ok {
+		return emoji
+	}
+
+	emojiToParse := strings.Split(emojiVal, "-")
+	parsedEmoji := ""
+	for _, val := range emojiToParse {
+		parsedVal, err := strconv.ParseInt(val, 16, 32)
+		if err != nil {
+			l4g.Error(err.Error())
+			return emoji
+		}
+		parsedEmoji = parsedEmoji + string(parsedVal)
+	}
+	return parsedEmoji
+}
+
+func ReplaceTextWithEmoji(text string) string {
+	regex := regexp.MustCompile("(:\\w+:)")
```

Comment:
Instead of `\w`. we need to use `[a-zA-Z0-9_-]` since emojis can contain hyphens. Also, this should be able to match `:+1:` and `:-1:`, so the full pattern could be `(:(?:[a-zA-Z0-9_-]+|\\+1|-1):)` to cover all those